### ISSUE TITLE
validation_set doesn't seem to run through Image Augmentation

### ISSUE THREAD

**Issue by tetmin** [tflearn/tflearn#215]
I'm trying to use validation_set=0.1 with the below tflearn image augmentation pipeline, but I get the following error which makes it seem like tflearn is trying to run the metric on the original image (128,128,3) instead of an augmented one (112,112,3):

Cannot feed value of shape (128, 128, 128, 3) for Tensor u'InputData/X:0', which has shape '(?, 112, 112, 3)'

```python
img_aug = ImageAugmentation()
img_aug.add_random_crop((112,112))

net = input_data(shape=[None, 112, 112, 3], data_augmentation=img_aug)
...
...
net = regression(net, optimizer='adam', loss='mean_square', learning_rate=0.001)

model = tflearn.DNN(net, tensorboard_verbose=0)
model.fit(X, latent, n_epoch=50, shuffle=True, show_metric=True,
          batch_size=128, validation_set=0.1)
```

**Comment by aymericdamien** [comment]
Ah yes, data augmentation is supposed to only be applied during training time, but that may be an issue here. To fix that, you could either use padding, so that random crop conserve original dimensions, otherwise, you can manually implement add_random_crop inside DataPreprocessing (that will apply that function at both training and testing time).
Nevertheless, we should think about an official way to handle such particular case.

**Comment by tetmin** [comment]
Thanks, I used padding to get it working in the meantime but would prefer to use 112 since my stacked convnets are scaled to give a nicely sized output. When you say manually implement, you mean re-write the add_random_crop function in python or is there a way to attach tensorflow pipelines & queues to tflearn?

If I do this will it preserve the queuing & out-of-core functionality in tflearn for the hdf5 file i'm using as an input?

**Comment by aymericdamien** [comment]
Yes, nothing will changed, you just have to basically copy/paste https://github.com/tflearn/tflearn/blob/master/tflearn/data_augmentation.py#L70 and https://github.com/tflearn/tflearn/blob/master/tflearn/data_augmentation.py#L160 into 'ImagePreprocessing' class (https://github.com/tflearn/tflearn/blob/master/tflearn/data_preprocessing.py#L301). Then ImagePreprocessing will support random_crop.
